### DISCUSSION TITLE
Archiving a directory for sinking to a server

### DISCUSSION THREAD

**Discussion by petermp79** [vectordotdev/vector#16498] [Q&A]
Can vector agent archive a directory and sink it off to a server using either socket or websocket? Any hints?

Thanks

**Comment by StephenWakely** [comment]
Do you mean read all the files in a director using the [file source](https://vector.dev/docs/reference/configuration/sources/file/) and sending the data over a [socket sink](https://vector.dev/docs/reference/configuration/sinks/socket/)?

That is possible, but the shape of the data will determine how successful it is. Vector is designed to read log files line by line. If the file is a line delimited log file you should be good, reading a binary file or one with lines that are too long will not work very well.
